llvm/lib/IR/IntrinsicInst.cpp

```diff
@@ -136,6 +137,8 @@ void DbgVariableIntrinsic::replaceVariableLocationOp(Value *OldValue,
   auto Locations = location_ops();
   auto OldIt = find(Locations, OldValue);
   if (OldIt == Locations.end()) {
+    if (AllowEmpty)
+      return;
```

Comment:
Why is this needed now (can you point to a test that fails without it)?

Reply:
The included test fails:
`llvm/test/Transforms/CallSiteSplitting/callsite-split-debug.ll`

This is the line that triggers the assertion:
`  call void @llvm.dbg.value(metadata !DIArgList(i16 %c, i16 %c), metadata !7, metadata !DIExpression()), !dbg !8
`

Reply:
Also, looking at the `DbgVariableRecord::replaceVariableLocationOp` the condition should be:
```
    if (AllowEmpty || DbgAssignAddrReplaced)
      return;
```

Reply:
Ah I see, yeah that makes sense. Thanks!